build_tools/log.py

```diff
@@ -15,6 +15,15 @@
 import sys
 
 def log(msg):
-    print msg
+    if type(msg) == bytes:
+        try:
+            print (msg.decode('utf-8'))
+        except:
+            try:
+                print (msg.decode('utf-16'))
+            except:
+                pass
+    else:
+        print(msg)
```

Comment:
In python3, there is a stricter distinction between encoded strings and binary strings, so we need to take care of all those cases.